py_zipkin/_encoding_helpers.py

```diff
@@ -1,12 +1,163 @@
 import socket
 from collections import namedtuple
 
+from enum import Enum
+
+from py_zipkin.exception import ZipkinError
+
+
 Endpoint = namedtuple(
     'Endpoint',
     ['service_name', 'ipv4', 'ipv6', 'port'],
 )
 
 
+_V1Span = namedtuple(
+    'V1Span',
+    ['trace_id', 'name', 'parent_id', 'id', 'timestamp', 'duration', 'endpoint',
+     'debug', 'annotations', 'binary_annotations', 'sa_endpoint'],
+)
+
+
+class Kind(Enum):
+    CLIENT = 'CLIENT'
+    SERVER = 'SERVER'
+    LOCAL = None
+
+
+_DROP_ANNOTATIONS_BY_KIND = {
+    Kind.CLIENT: {'ss', 'sr'},
+    Kind.SERVER: {'cs', 'cr'},
+}
+
+
+class Span(object):
+    def __init__(
+        self,
+        trace_id,
+        name,
+        parent_id,
+        span_id,
+        timestamp,
+        duration,
+        annotations,
+        tags,
+        kind,
+        local_endpoint=None,
+        service_name=None,
+        sa_endpoint=None,
+        include=None,
+        report_timestamp=True,
+    ):
+        """Internal Span type.
+
+        It doesn't exactly map to either V1 or V2, since an
+        intermediate format makes it way easier to convert
+        to either format.
+
+        :param trace_id: Trace id.
+        :type trace_id: str
+        :param name: Name of the span.
+        :type name: str
+        :param parent_id: Parent span id.
+        :type parent_id: str
+        :param span_id: Span id.
+        :type span_id: str
+        :param timestamp: start timestamp in seconds.
+        :type timestamp: float
+        :param duration: span duration in seconds.
+        :type duration: float
+        :param annotations: Optional dict of str -> timestamp annotations.
+        :type annotations: dict
+        :param tags: Optional dict of str -> str span tags.
+        :type tags: dict
+        :param kind: Span type (client, server, local, etc...)
+        :type kind: Kind
+        :param local_endpoint: The host that recorded this span.
+        :type local_endpoint: Endpoint
+        :param service_name: The name of the called service
+        :type service_name: str
+        :param sa_endpoint: Remote server in client spans.
+        :type sa_endpoint: Endpoint
+        :param include: Which annotations to include
+        :type include: iterable
+        :param report_timestamp: Whether the span should report
+            timestamp and duration.
```

Comment:
I generate `cr`, `cs`, `sr`, `ss` annotations inside `Span`, so I need to propagate the timestamp and duration.

Since those annotations are V1 specific I think this is a better alternative.